release/build.gradle

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: 'groovy'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile 'org.codehaus.groovy:groovy-all:2.4.13'
+    compile 'commons-cli:commons-cli:1.2'
+}
+
+// Nightly releases
+def verNightly = project.findProperty('ver') ?: '2.4.0-SNAPSHOT'
+def repoNightly = project.findProperty('repourl') ?: 'https://repository.apache.org/content/repositories/snapshots'
+def argsNightly = ["--ver=${verNightly}", "--repourl=${repoNightly}"]
+
+// External releases
+def verRelease = project.findProperty('ver') ?: '2.3.0'
+def repoRelease = project.findProperty('repourl') ?: 'https://repository.apache.org/content/repositories/orgapachebeam-1027'
+def argsRelease = ["--ver=${verRelease}", "--repourl=${repoRelease}"]
+
+// For dataflow runner
+def cloudProject = project.findProperty('cloudProject') ?: 'apache-beam-testing'
+def cloudBucket = project.findProperty('cloudBucket') ?: 'temp-storage-for-release-validation-tests/quickstart'
+def argsProject = ["--project=${cloudProject}", "--bucket=${cloudBucket}"]
+
+// Generates tasks for running each quickstart in each configuration, like
+// :runQuickstartJavaDirectNightly
+// :runQuickstartJavaDirectRelease
+// :runQuickstartJavaDataflowNightly
+// :runQuickstartJavaDataflowRelease
+for (String release : ["Release", "Nightly"]) {
+  for (String runner : ["Direct", "Dataflow", "Apex", "Spark", "FlinkLocal"]) {
+    def taskName = "runQuickstartJava${runner}${release}"
+    def argsNeeded =  release == "Nightly" ?  argsNightly : argsRelease
+    if (runner == "Dataflow") {
+      argsNeeded += argsProject
+    } 
+    println "Generating :${taskName}"
+    task "${taskName}" (dependsOn: 'classes', type: JavaExec) {
```

Comment:
In this case, it would be good to have these tasks defined in the individual runners projects instead of aggregated all here.

You can add a method to `build_rules.gradle` which is like `createJavaQuickstartValidationTask` and then call it in spark/dataflow/apex/...

Reply:
Done. Added the tasks in each runner build.gradle, calling createJavaQuickstartValidationTask from build_rules.gradle